packages/core/src/test/amazonqFeatureDev/session/session.test.ts

```diff
@@ -31,4 +40,77 @@ describe('session', () => {
             })
         })
     })
+    describe('insertChanges', async () => {
+        const controllerSetup: ControllerSetup = await createController()
+
+        const workspaceFolder = controllerSetup.workspaceFolder
+
+        beforeEach(async () => {
+            const uploadID = '789'
+            const tabID = '123'
+            const workspaceFolders = [controllerSetup.workspaceFolder] as CurrentWsFolders
+
+            sinon.stub(VirtualFileSystem.prototype)
+
+            const testConfig = {
+                conversationId: conversationID,
+                proxyClient: {
+                    createConversation: () => sinon.stub(),
+                    createUploadUrl: () => sinon.stub(),
+                    generatePlan: () => sinon.stub(),
+                    startCodeGeneration: () => sinon.stub(),
+                    getCodeGeneration: () => sinon.stub(),
+                    exportResultArchive: () => sinon.stub(),
+                } as unknown as FeatureDevClient,
+                sourceRoots: [''],
+                uploadId: uploadID,
+                workspaceFolders,
+            }
+            const testApproach = 'test-approach'
+
+            const codeGenState = new CodeGenState(
+                testConfig,
+                testApproach,
+                [
+                    {
+                        zipFilePath: 'myfile.js',
+                        relativePath: 'myfile.js',
+                        fileContent: '',
+                        rejected: false,
+                        virtualMemoryUri: '' as unknown as vscode.Uri,
+                        workspaceFolder,
+                    },
+                    {
+                        zipFilePath: 'myfile_1.js',
+                        relativePath: 'myfile_1.js',
+                        fileContent: '',
+                        rejected: true,
+                        virtualMemoryUri: '' as unknown as vscode.Uri,
+                        workspaceFolder: controllerSetup.workspaceFolder,
+                    },
+                ],
+                [],
+                [],
+                tabID,
+                0
+            )
+            session = await createSession({ messenger, sessionState: codeGenState, conversationID })
+        })
+        afterEach(() => {
+            sinon.restore()
+        })
+        it('only insert non rejected files', async () => {
+            const fsStub = sinon.stub(FileSystemCommon.instance, 'writeFile').resolves()
+
+            await session.insertChanges()
+
+            await waitUntil(() => {
+                return Promise.resolve(fsStub.callCount > 0)
+            }, {})
+
+            const absolutePath = path.join(workspaceFolder.uri.fsPath, 'myfile.js')
+            assert.ok(fsStub.calledOnce)
+            assert.ok(fsStub.calledWith(absolutePath, ''))
```

Comment:
Is this empty string here the file content?

It'd make it clearer if we added some value to the file content on the files above.
`myFile.js` could have a content of `Contents of myFile.js` for example and this could be tested here. Better than testing for an empty string :)

Reply:
added!